src/backend/query_builder.rs

```diff
@@ -1512,3 +1513,45 @@ impl QuotedBuilder for CommonSqlQueryBuilder {
 impl EscapeBuilder for CommonSqlQueryBuilder {}
 
 impl TableRefBuilder for CommonSqlQueryBuilder {}
+
+pub(crate) fn common_inner_expr_well_known_greater_precedence(
+    inner: &SimpleExpr,
+    outer_oper: &Oper,
+) -> bool {
+    match inner {
+        SimpleExpr::Column(_) => true,
+        SimpleExpr::Tuple(_) => true,
+        SimpleExpr::Constant(_) => true,
+        SimpleExpr::FunctionCall(_) => true,
+        SimpleExpr::AsEnum(_, _) => true,
+        SimpleExpr::Value(_) => true,
+        SimpleExpr::Keyword(_) => true,
+        SimpleExpr::SubQuery(_, _) => true,
+        SimpleExpr::Binary(_, inner_oper, _) => {
+            let inner_oper: Oper = (*inner_oper).into();
+            if inner_oper.is_arithmetic() || inner_oper.is_shift() {
```

Comment:
A common way to represent operator precedence is with integer levels, then you can do a lookup in a table and a straight comparison.

Would that approach work here? Or are there cases where it's not a total order? (E.g. lacking transitivity, maybe around the between operator?)

Reply:
Yes, there is a partial ordering here. SQL parsers disagree about the finer details, and my goal was to represent those precedence relations that are well known, where adding unnecessary parentheses would create cluttered SQL. 

I thought just representing the relation directly would be easier to read and think about, and gave me an easier time to try to recreate most current behaviour.